app/models/game.rb

```diff
@@ -16,14 +16,35 @@ def in_check?
     %w(white black).each do |king_color|
       king = pieces.find_by(type: 'King', color: king_color)
 
-      enemy_color =  %w(white black).select { |color| king_color != color }
+      enemy_color = %w(white black).select { |color| king_color != color }
 
       return king if location_is_under_attack_by_color?(enemy_color, king.row_coordinate, king.column_coordinate)
     end
 
     nil
   end
 
+  def checkmate?
+    king = in_check?
+    return false if king.nil?
+
+    # scan entire board to collect required data
+    king_moves_list, attackers, friendly_list = build_attackers_and_friendly_lists(king)
+
+    # determine if King can move out of check
+    return false if king_can_move_out_of_check?(king, king_moves_list)
+
+    return true if attackers.count > 1
+
+    # determine if attacker can be captured
+    return false if king_attacker_can_be_captured?(attackers.first, friendly_list)
+
+    # determine if check can be blocked
+    return false if check_can_be_blocked?(king, attackers.first, friendly_list)
+
+    true
+  end
```

Comment:
Reads very easily. I like how you extracted complexity into well named private methods. :+1: